sequencer/src/block2.rs

```diff
@@ -108,92 +230,85 @@ impl BlockPayload {
             },
             namespace_table,
         ))
+        .context(BlockBuildingSnafu)
     }
 
-    fn from_bytes<B>(bytes: B) -> Self
+    // TODO(746) from_bytes doesn't need `metadata`!
+    fn from_bytes<I>(encoded_transactions: I, _metadata: &Self::Metadata) -> Self
     where
-        B: IntoIterator<Item = u8>,
+        I: Iterator<Item = u8>,
     {
-        Self {
-            payload: bytes.into_iter().collect(),
-            tx_table_len_proof: Default::default(),
-        }
+        Self::from_bytes(encoded_transactions)
```

Comment:
can we eliminate the delegation to `Self::from_bytes` and simply put that function body in here?

Reply:
...just like you did for `from_transactions`?

Reply:
yes, I think it is doable. But seems like our `malformed_payloads`-test needs changing since we need to make the metadata (ns_table) available.